### PR TITLE
[MOB-3051] Loyalty Balance - Check description

### PR DESCRIPTION
**Pull request by EdwardW2**
## JIRA
[MOB-3051](https://karhoo.atlassian.net/browse/MOB-3051)

## DESCRIPTION
Basic Implementation of the Loyalty Balance with its own service for now. May be modified in the future to be part of Payments or Trip service instead but has been decided to only do a very basic and unused implementation for now.

Integration tests for Loyalty ignored for now due to unknown failures. All objects and errors coming back as Null even though tests are similar to all other integration tests???

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [x] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [x] Other: NSDK Improvements

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [x] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [x] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: sdk/src/androidTest/java/com/karhoo/sdk/api/LoyaltyIntegrationTest.kt [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+     * Then:    The response payload should be valid
+     **/
+    @Test
+    fun loyaltyBalanceSuccess() {
+        serverRobot {
+            getLoyaltyBalanceResponse(code = HTTP_OK, response = LOYALTY_BALANCE)
+        }
+        var result: LoyaltyBalance? = LoyaltyBalance(123, false)
+
+        KarhooApi.loyaltyService.getBalance(ServerRobot.LOYALTY_ID).execute {
```
Comment (jsm20) [inline sdk/src/androidTest/java/com/karhoo/sdk/api/LoyaltyIntegrationTest.kt:62]: Static imports

File: sdk/src/androidTest/java/com/karhoo/sdk/api/VerifyQuoteIntegrationTest.kt [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+package com.karhoo.sdk.api
 
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import com.github.tomakehurst.wiremock.junit.WireMockRule
-import com.karhoo.sdk.api.KarhooApi
-import com.karhoo.sdk.api.KarhooError
 import com.karhoo.sdk.api.model.Quote
 import com.karhoo.sdk.api.network.response.Resource
 import com.karhoo.sdk.api.testrunner.SDKTestConfig
+import com.karhoo.sdk.api.util.ServerRobot
```
Comment (jsm20) [inline sdk/src/androidTest/java/com/karhoo/sdk/api/VerifyQuoteIntegrationTest.kt:8]: Duplicated

Comment (EdwardW2) [reply sdk/src/androidTest/java/com/karhoo/sdk/api/VerifyQuoteIntegrationTest.kt:8]: Tried fixing this already Android Studio throws errors if I try to remove these

File: sdk/src/main/java/com/karhoo/sdk/api/service/loyalty/LoyaltyInteractor.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import kotlinx.coroutines.Dispatchers.Main
+import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
+
+internal class LoyaltyInteractor @Inject constructor(credentialsManager: CredentialsManager,
+                                                    private val apiTemplate: APITemplate,
+                                                    context: CoroutineContext = Main) :
+        BaseCallInteractor<LoyaltyBalance>(true, credentialsManager, apiTemplate, context) {
+
+            internal var loyaltyId: String? = null
```
Comment (jsm20) [inline sdk/src/main/java/com/karhoo/sdk/api/service/loyalty/LoyaltyInteractor.kt:20]: Formatting

File: sdk/src/androidTest/java/com/karhoo/sdk/api/LoyaltyIntegrationTest.kt (hunk 1 above)
Comment (EdwardW2) [reply sdk/src/androidTest/java/com/karhoo/sdk/api/LoyaltyIntegrationTest.kt:62]: Not sure what is being referred to here exactly, but again, Android studio newest version does not like it when I try to remove some static imports in the project.

File: sdk/src/test/java/com/karhoo/sdk/api/service/loyalty/LoyaltyInteractorTest.kt [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        verify(apiTemplate).getLoyaltyBalance(LoyaltyInteractorTest.LOYALTY_ID)
+    }
+
+    /**
+     * Given:   A valid loyalty id
+     * When:    When requesting loyalty balance
+     * Then:    The loyalty balance should return as burnable.
+     */
+    @Test
+    fun `requesting loyalty balance returns burnable value`() {
```
Comment (jsm20) [inline sdk/src/test/java/com/karhoo/sdk/api/service/loyalty/LoyaltyInteractorTest.kt:70]: Is this test doing anything different to the one above?

File: sdk/src/androidTest/java/com/karhoo/sdk/api/VerifyQuoteIntegrationTest.kt (hunk 2 above)
Comment (EdwardW2) [reply sdk/src/androidTest/java/com/karhoo/sdk/api/VerifyQuoteIntegrationTest.kt:8]: Both server robots here are different, not duplicated

**Review by jsm20 — APPROVED** [review]

**Pull request merged**